test/cpp_extensions/setup.py

```diff
@@ -59,5 +59,5 @@
     name='torch_test_cpp_extension',
     packages=['torch_test_cpp_extension'],
     ext_modules=ext_modules,
-    include_dirs='self_compiler_include_dirs_test',
+    include_dirs='self_compiler_include_dirs_test with spaces',
```

Comment:
Looks like the no-ninja path doesn't support having spaces although it sometimes works (paths with spaces work on my machine).

```
May 20 02:44:48 gcc -pthread -B /opt/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/conda/lib/python3.6/site-packages/torch/include -I/opt/conda/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -I/opt/conda/lib/python3.6/site-packages/torch/include/TH -I/opt/conda/lib/python3.6/site-packages/torch/include/THC -Iself_compiler_include_dirs_test with spaces -I/opt/conda/include/python3.6m -c extension.cpp -o build/temp.linux-x86_64-3.6/extension.o -g -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=cpp -D_GLIBCXX_USE_CXX11_ABI=1 -std=c++14
May 20 02:44:48 gcc: error: with: No such file or directory
May 20 02:44:48 gcc: error: spaces: No such file or directory
May 20 02:44:48 error: command 'gcc' failed with exit status 1
May 20 02:44:48 Traceback (most recent call last):
May 20 02:44:48   File "test/run_test.py", line 703, in <module>
May 20 02:44:48     main()
May 20 02:44:48   File "test/run_test.py", line 696, in main
May 20 02:44:48     raise RuntimeError(message)
May 20 02:44:48 RuntimeError: test_cpp_extensions_aot_no_ninja failed!
```
(https://circleci.com/gh/pytorch/pytorch/5530602?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

We should keep both `self_compiler_include_dirs_test` and `self_compiler_include_dirs_test with spaces` and switch between them depending on `USE_NINJA`. Also it would be nice to have a comment here, something along the lines of "test that the ninja build path supports directories with spaces. The no-ninja path doesn't support this".